ddtrace/appsec/_handlers.py

```diff
@@ -95,8 +95,11 @@ async def _on_asgi_request_parse_body(receive, headers):
         data_received = await receive()
         body = data_received.get("body", b"")
 
-        async def receive():
-            return data_received
+        async def receive_wrapped(once=[True]):
+            if once[0]:
```

Comment:
I wonder if we could store the state somewhere else less tricky

Reply:
This is the equivalent to a static local variable in python. Otherwise we can use a nonlocal variable.